itest/lnd_sweep_test.go

```diff
@@ -800,6 +800,433 @@ func testSweepHTLCs(ht *lntest.HarnessTest) {
+	//
+	// Prepare node params.
+	cfg := []string{
+		"--protocol.anchors",
+		fmt.Sprintf("--sweeper.nodeadlineconftarget=%v", deadlineA),
+		fmt.Sprintf("--bitcoin.defaultremotedelay=%v", toLocalCSV),
+	}
+	openChannelParams := lntest.OpenChannelParams{
+		Amt:     fundAmt,
+		PushAmt: bobBalance,
+	}
+
+	// Create a two hop network: Alice -> Bob.
+	chanPoints, nodes := createSimpleNetwork(ht, cfg, 2, openChannelParams)
+
+	// Unwrap the results.
+	chanPoint := chanPoints[0]
+	alice, bob := nodes[0], nodes[1]
+
+	invoice := &lnrpc.Invoice{
+		Memo:       "bob",
+		Value:      htlcAmt,
+		CltvExpiry: finalCltvDelta,
+	}
+	resp := bob.RPC.AddInvoice(invoice)
+
+	// Send a payment with a specified finalCTLVDelta, and assert it's
+	// succeeded.
+	req := &routerrpc.SendPaymentRequest{
+		PaymentRequest: resp.PaymentRequest,
+		TimeoutSeconds: 60,
+		FeeLimitMsat:   noFeeLimitMsat,
+	}
+	ht.SendPaymentAssertSettled(alice, req)
+
+	// Assert Alice's to_remote (Bob's to_local) output is the htlc amount.
+	ht.AssertChannelLocalBalance(bob, chanPoint, htlcAmt)
+	bobToLocal := htlcAmt
+
+	// Get Alice's channel to calculate Alice's to_local output amount.
+	aliceChan := ht.GetChannelByChanPoint(alice, chanPoint)
+
+	// TODO(yy): It looks like the funder is paying the fundee's anchor?
+	expectedToLocal := int64(fundAmt) - aliceChan.CommitFee - htlcAmt -
+		330*2
+
+	// Assert Alice's to_local output is correct.
+	aliceToLocal := aliceChan.LocalBalance
+	require.EqualValues(ht, expectedToLocal, aliceToLocal)
+
+	// Alice force closes the channel.
+	ht.CloseChannelAssertPending(alice, chanPoint, true)
+
+	// Now that the channel has been force closed, it should show up in the
+	// PendingChannels RPC under the waiting close section.
+	ht.AssertChannelWaitingClose(alice, chanPoint)
+
+	// We should see neither Alice or Bob has any pending sweeps as there
+	// are no time-sensitive HTLCs.
+	ht.AssertNumPendingSweeps(alice, 0)
+	ht.AssertNumPendingSweeps(bob, 0)
+
+	// Mine a block to confirm Alice's force closing tx. Once it's
+	// confirmed, we should see both Alice and Bob's anchors being offered
+	// to their sweepers.
+	ht.MineBlocksAndAssertNumTxes(1, 1)
+
+	// Alice should have one pending sweep,
+	// - anchor sweeping from her local commitment.
+	ht.AssertNumPendingSweeps(alice, 1)
+
+	// Bob should have two pending sweeps,
+	// - anchor sweeping from the remote anchor on Alice's commit tx.
+	// - commit sweeping from the to_remote on Alice's commit tx.
+	ht.AssertNumPendingSweeps(bob, 2)
+
+	// Mine one more empty block should trigger Bob's sweeping. Since we
+	// use a CSV of 3, this means Alice's to_local output is one block away
+	// from being mature.
+	ht.MineEmptyBlocks(1)
+
+	// We expect to see one sweeping tx in the mempool:
+	// - Alice's anchor sweeping tx must have been failed due to the fee
+	//   rate chosen in this test - the anchor sweep tx has no output.
+	// - Bob's sweeping tx, which sweeps both his anchor and commit outputs.
```

Comment:
we should not bundle anchors with normal outputs for the reasons stated here: https://github.com/lightningnetwork/lnd/pull/8514#issuecomment-2042758923

Reply:
18.1